src/common/search/SearchService.js

```diff
@@ -228,14 +242,14 @@
       searchlayer_.getSource().clear();
       mapService_.map.removeLayer(searchlayer_);
       mapService_.map.addLayer(searchlayer_);
-      forEachArrayish(results, function(result) {
+      forEachArrayish(results, function(result, index) {
         var olFeature = new ol.Feature();
         // properly assign feature data (removing hashkey metadata)
         olFeature.properties = angular.copy(result);
         if (result.name.length > 25) {
-          olFeature.setId(result.name.substring(0, 25) + '...');
+          olFeature.setId(result.name.substring(0, 25) + '...' + index);
         } else {
-          olFeature.setId(result.name);
+          olFeature.setId(result.name + index);
```

Comment:
Added the use of index because the MIDB search results do not have unique names despite being different data points, so this is done "just in case" to ensure the feature ID is unique.